### PR TITLE
Package.json updates: Windows & yarn

### PR DESCRIPTION
**Pull request by Wick**
Some improvements (mostly) for Windows developers:

Expanded list of supported engines to include versions up to the latest version of node (gives better support for yarn package manager as a bonus).

Escape quotes in the ```npm run format``` command. This will make the command run without error in Windows environments (should be fine for other OSes as well I think?) . 

Added lockfile for yarn to get consistent installs across machines. Reference: https://yarnpkg.com/lang/en/docs/yarn-lock/

### CODE REVIEW

**Comment by NetOperatorWibby** [comment]
@Wick Could you update the Installation section of the README with yarn instructions? I don't personally use it.

File: package.json [hunk 1]
```diff
@@ -69,7 +69,7 @@
     "updates": "^6.2.1"
   },
   "engines": {
-    "node": "10.2.x"
+     "node": ">=10.2 <10.15.1"
```
Comment (NetOperatorWibby) [inline package.json]: This looks confusing, what does it do?

File: package.json [hunk 2]
```diff
@@ -81,7 +81,7 @@
   "private": true,
   "scripts": {
     "css": "sass --load-path=node_modules --update app/sass:app/dist --style compressed",
-    "format": "eslint '**/*.js' --fix --ignore-pattern '/app/dist/'",
+    "format": "eslint \"**/*.js\" --fix --ignore-pattern '/app/dist/'",
```
Comment (NetOperatorWibby) [inline package.json]: Does the second command in this line need to escaped too?

Comment (Wick) [reply package.json]: Not needed, runs just fine after escaping only the eslint command. :)

File: package.json (hunk 1 above)
Comment (Wick) [reply package.json]: For regular users using npm this is just an indicator on which versions of node that can run this project fine.
If you use an alternative dependency manager, like Yarn, you'll get an error if your node version doesn't match the ones listed under "engines".
My change only extends the list of working node versions to include the versions of node between the last set version(10.2.x) and the latest recommended version listed on https://nodejs.org/en/ (which I use and it works without problems).

Not really an important change, want me to revert it?

**Comment by Wick** [comment]
Updated the readme in my latest commit, let me know if I can improve it in any way.  :)

I know npm supports lock-files as well nowadays, but noticed it is explicitly set to not generate one in [.npmrc](https://github.com/lbryio/lbry.tech/blob/master/.npmrc). So I only mentioned the lock-file for Yarn for now.

**Comment by NetOperatorWibby** [comment]
That's cool. I've had more negative than positive experiences with lockfiles so I disabled them.

**Pull request merged**